File: ai-kore/app/services/crawl_service.py
"""
爬虫 / 图片处理业务逻辑

职责：
- 编排单张图片处理管线
- 批量时串行处理（每张处理完再处理下一张）
"""

from typing import Any, List

from pipeline.image_pipeline import process_images_batch, process_single_image


def process_one_image(url: str) -> dict[str, Any]:
    """
    处理单张图片：下载 → OSS → CLIP → OCR → 向量化 → Milvus。

    Args:
        url: 图片 URL

    Returns:
        结构化结果字典
    """
    return process_single_image(url)


def process_many_images(urls: List[str]) -> List[dict[str, Any]]:
    """
    批量处理图片，串行执行（每张处理完再处理下一张）。

    Args:
        urls: 图片 URL 列表

    Returns:
        每张图片的结果列表
    """
    return process_images_batch(urls)

File: smart_meter/src/main/java/com/purelearning/smart_meter/service/CrawlService.java
package com.purelearning.smart_meter.service;

import com.purelearning.smart_meter.dto.crawl.CrawlProcessImageResponse;
import com.purelearning.smart_meter.dto.crawl.CrawlProcessImagesResponse;

import java.util.List;

/**
 * 图片离线管线业务接口。
 * 调用 ai-kore /api/v1/crawl/process-image，将单张图片 URL 交给 Python 端完成下载→OSS→CLIP→OCR→Milvus→MySQL。
 */
public interface CrawlService {

    /**
     * 提交单张图片 URL 到离线管线。
     *
     * @param url 图片公网 URL
     * @return 处理结果（包含 imageUrl、embeddingId 等）
     */
    CrawlProcessImageResponse processImage(String url);

    /**
     * 批量提交图片 URL，串行处理；转发 ai-kore POST /api/v1/crawl/process-images。
     *
     * @param urls 图片直链列表（非空）
     * @return 每张结果及成功数量汇总
     */
    CrawlProcessImagesResponse processImages(List<String> urls);

    /**
     * 从网页 HTML 中解析图片 URL，不直接入库。
     *
     * @param pageUrl 网页 URL
     * @param limit   最多返回数量
     * @return 去重后的图片直链列表
     */
    List<String> extractImageUrls(String pageUrl, int limit);
}

